core/services/registrysyncer/syncer.go

```diff
@@ -150,14 +166,36 @@ func (s *registrySyncer) syncLoop() {
 			return
 		case <-ticker.C:
 			s.lggr.Debug("starting regular sync with the remote registry")
-			err := s.sync(ctx)
+			err := s.sync(ctx, false)
 			if err != nil {
 				s.lggr.Errorw("failed to sync with remote registry", "error", err)
 			}
 		}
 	}
 }
 
+func (s *registrySyncer) updateStateLoop() {
+	ctx, cancel := s.stopCh.NewCtx()
+	defer cancel()
+
+	for {
+		select {
+		case <-s.stopCh:
+			return
+		case localRegistry, ok := <-s.updateChan:
+			if !ok {
+				// channel has been closed, terminating.
+				return
+			}
+			s.dbMu.Lock()
```

Comment:
This lock here is a bit odd: what are you trying to guard against?

I only see one Lock, and this loop should be running as a singleton so there would be no concurrent access; furthermore I'd expect the locking to happen in the store (either via means of an appropriate transaction/for update statement or mutex inside the database itself, since you would always call the mutex + orm together)